ipc/common/messagesFactory.go

```diff
@@ -0,0 +1,106 @@
+package common
+
+import (
+	"fmt"
+)
+
+// CreateMessage creates a message given its kind
+func CreateMessage(kind MessageKind) MessageHandler {
+	var message MessageHandler
+
+	switch kind {
```

Comment:
would a map be faster ? or a slice instead of switch ?

Reply:
Fixed, using a slice.